src/main/java/codeu/controller/ProfileServlet.java

```diff
@@ -13,37 +14,45 @@
 
 import codeu.model.data.User;
 import codeu.model.store.basic.UserStore;
+import codeu.model.store.basic.MessageStore;
 
 public class ProfileServlet extends HttpServlet {
 
   /** Store class that gives access to Users. */
   private UserStore userStore;
 
   /**
-   * Set up state for handling registration-related requests. This method is only called when
-   * running in a server, not when running in a test.
-   */
+  * Set up state for handling registration-related requests. This method is only called when
+  * running in a server, not when running in a test.
+  */
   @Override
   public void init() throws ServletException {
     super.init();
     setUserStore(UserStore.getInstance());
   }
 
   /**
-   * Sets the UserStore used by this servlet. This function provides a common setup method for use
-   * by the test framework or the servlet's init() function.
-   */
+  * Sets the UserStore used by this servlet. This function provides a common setup method for use
+  * by the test framework or the servlet's init() function.
+  */
   void setUserStore(UserStore userStore) {
     this.userStore = userStore;
   }
 
   /**
-   * This function fires when a user requests the /login URL. It simply forwards the request to
-   * profile.jsp.
-   */
+  * This function fires when a user requests the /profile URL. It simply forwards the request to
+  * profile.jsp.
+  */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
-      throws IOException, ServletException {
+  throws IOException, ServletException {
+    String requestUrl = request.getRequestURI();
+    String usernameUrl = requestUrl.substring("/users/".length());
```

Comment:
It looks like you're trying to extract the username from the URL. If that is the case, you might want to get the index of "/users/" first and use that in the substring. Otherwise "/users/".length() can be replaced by 7.